contracts/src/HyperlaneValidatorBlueprint.sol

```diff
@@ -1,19 +1,199 @@
 // SPDX-License-Identifier: UNLICENSE
 pragma solidity >=0.8.13;
 
-import "tnt-core/BlueprintServiceManagerBase.sol";
+import "./ChallengerEnrollment.sol";
 
 /**
  * @title HyperlaneValidatorBlueprint
  * @dev This contract is a blueprint for a Hyperlane validator deployment.
+ * It supports integration with challenger contracts for verifiable fraud proofs and slashing.
+ * The blueprint itself acts as the service manager in Tangle's architecture.
  */
-contract HyperlaneValidatorBlueprint is BlueprintServiceManagerBase {
+contract HyperlaneValidatorBlueprint is ChallengerEnrollment {
+    struct HyperlaneRequestInputs {
+        address[] challengers;
+        uint32 originDomain;
+        uint32[] destinationDomains;
+    }
+
+    // Structure to store domain information for services
+    struct ServiceDomains {
+        uint32 originDomain;
+        uint32[] destinationDomains;
+        address owner;
+        address[] permittedCallers;
+        uint64 ttl;
+    }
+
+    // Structure to store approved operator information
+    struct OperatorApproval {
+        address operator;
+        bytes publicKey;
+        bool isApproved;
+    }
+
+    // Mapping of serviceId to domain information
+    mapping(uint256 => ServiceDomains) public serviceDomains;
+
+    // Mapping of requestId to request inputs
+    mapping(uint64 => HyperlaneRequestInputs) private _pendingRequests;
+
+    // Mapping of requestId to approved operators
+    mapping(uint64 => OperatorApproval[]) private _approvedOperators;
+
+    // Event emitted when a service instance is created
+    event ServiceInstanceCreated(
+        uint256 indexed serviceId,
+        address[] challengers,
+        uint32 originDomain,
+        address owner,
+        address[] permittedCallers,
+        uint64 ttl
+    );
+
+    // Event emitted when an operator approves a service request
+    event OperatorApproved(
+        uint64 indexed requestId,
+        address indexed operator,
+        bytes publicKey
+    );
+
+    // Event emitted when an operator rejects a service request
+    event OperatorRejected(
+        uint64 indexed requestId,
+        address indexed operator
+    );
+
+    /**
+     * @notice Handles incoming service requests
+     * @dev Stores request parameters for later processing during service initialization
+     * @param params The request parameters
+     */
+    function onRequest(ServiceOperators.RequestParams calldata params) override external payable virtual onlyFromMaster {
+        // Decode and store the blueprint-specific request inputs
+        HyperlaneRequestInputs memory blueprintInputs = abi.decode(params.requestInputs, (HyperlaneRequestInputs));
```

Comment:
`requestInputs` will always be empty, that's what's causing the revert.

`request_params` needs to be specified in the `blueprint!` macro in the binary:

https://github.com/tangle-network/hyperlane-validator-blueprint/blob/5f18a14a3676c8ba9f136ad6153a67b53680b4b6/hyperlane-validator-bin/build.rs#L15-L20